score.go

```diff
@@ -320,6 +354,14 @@ func (ps *peerScore) background(ctx context.Context) {
 
 // inspectScores dumps all tracked scores into the inspect function.
 func (ps *peerScore) inspectScores() {
+	if ps.inspect != nil {
```

Comment:
The `WithPeerScoreInspect` option could be called twice, once with a simple inspector, once with an extended inspector. That wouldn't error and would lead to setting both inspectors, yet at runtime we would only call the simple one.

Either we should error on multiple `WithPeerScoreInspect` calls, or we should remove the `else` in this `if` block and turn it into a separate `if`, in order to call both inspectors.

Reply:
I think allowing multiple invocations of `WithPeerScoreInspect` might be a mistake because there are conflicts with the time period, let's disallow.
Regardless, I will remove the else as it is confusing.

Reply:
done.